### PR TITLE
refactor(style): update splash site

### LINKED ISSUE #25: Make the splash site look like bryan's mockup

### PR DESCRIPTION
**Pull request by johngruen**
Fixes #25

### CODE REVIEW

**Comment by johngruen** [comment]
@pdehaan getting on the subway. I expect this to be shot full of holes by the time I get back home :)

**Comment by johngruen** [comment]
fixes #25

File: routes/data.json [hunk 1]
```diff
@@ -3,22 +3,46 @@
...
+      "url": "https://www.mozilla.org/about/legal/"
+    },
+    {
+      "name": "cookies",
+      "url": "https://www.mozilla.org/about/legal/"
+    },
+    {
+      "name": "contribute",
+      "url": "https://www.mozilla.org/about/legal/"
+    }
```
Comment (meandavejustice) [inline routes/data.json]: All of these urls point to `/about/legal`

**Comment by meandavejustice** [comment]
I have pulled this down and visually this looks great, I feel like `p.disclaimer` could maybe use a tiny bit of `padding-top`.

**Comment by pdehaan** [comment]
This looks great @johngruen.
Only nits are minor and mostly unrelated (like the diagonal lines in the windows are the same for every window except the two buildings on the right). Or how we should move the inlined JavaScript code into main.js (so it can get bundled and minified -- and we should probably add sourcemaps).

We'll probably also want to minify the "normalize.css" and "tabzilla.css" as well. 

Nothing 404s, nothing in my dev console, new fonts look great.

Looking at Chrome, I do see an unsightly focus rect when entering my email address. And if I tab to the Sign up button (in Firefox), there is no indication that the submit button has focus (in Chrome I see an ugly focus rect, so at least keyboard users will know the button is "active").

**Comment by pdehaan** [comment]
I'm sorry @johngruen, you do great work! :eyes:

**Comment by pdehaan** [comment]
Here's one... inconsistent link UX.

The "Privacy Notice" link below the sign up form has an underline to imply a clickable piece of text.
The links at the bottom of the page (legal, privacy, cookies, contribute) don't have any text decorations or hover states (but both links DO have a nice, subtle focus rect when you tab to them).

I vote for at least some subtle hover states on the footer buttons.

**Comment by pdehaan** [comment]
Unrelated, but a nice conversation piece...

What is "Give us feedback"? Where are people expected to do that? Are we building a feedback form for this page/site? Or is that going to point to input.mozilla.org or Bugzilla (uh, no thanks), or GitHub issues (forcing people to be registered to GitHub to submit feedback/ideas -- limiting it to mostly developers only).

Let me know if you want me to spin this off to a separate bug. Or just feel free to delete this comment and we can move on.

**Comment by johngruen** [comment]
@pdehaan @meandavejustice addressed everything minus the window shadow direction...I'll let @bryanbell mess with that should he so desire.

Also added some new gulp task that you might like :)

File: views/includes/header.jade [hunk 2]
```diff
@@ -4,7 +4,6 @@ head
   meta(name="keywords" content="#{ keys }")
   meta(name="viewport" content="width=device-width, initial-scale=1")
   title= title
-  link(rel="stylesheet" href="//code.cdn.mozilla.net/fonts/fira.css")
-  link(rel="stylesheet" href="vendor/normalize.css/normalize.css")
-  link(rel="stylesheet" href="vendor/mozilla-tabzilla/css/tabzilla.css")
+
+  link(rel="stylesheet" href="http://fonts.googleapis.com/css?family=Open+Sans:400,300")
```
Comment (pdehaan) [inline views/includes/header.jade]: HTTPS please

File: views/layout.jade [hunk 3]
```diff
@@ -6,3 +6,4 @@ html(lang="en")
     block content
 
     include includes/footer
+    script(type="text/javascript" src="scripts/main.js")
```
Comment (pdehaan) [inline views/layout.jade]: We should use minimized file here. I still need to figures out source mapping for easier debugging.

File: src/styles/modules/_modules.scss [hunk 4]
```diff
@@ -1,2 +1,5 @@
+
```
Comment (pdehaan) [inline src/styles/modules/_modules.scss]: Nit: Remove blank line

File: .gitignore [hunk 5]
```diff
@@ -2,6 +2,7 @@
 components
```
Comment (pdehaan) [inline .gitignore]: Nit: remove components and .sass-cache. Bower and ruby sass artifacts.

**Comment by johngruen** [comment]
A hard lesson to learn in life is that it helps to look at WHY Travis is failing and not just flail around wildly.

# ¯\_(ツ)_/¯

File: gulpfile.js [hunk 6]
```diff
@@ -100,18 +104,37 @@ gulp.task('static-splash', function() {
...
+    'public',
+    'src/vendor'
+  ])
+})
+
+gulp.task('vendor', function() {
   runSequence(
+    'npm:tabzilla:css',
+    'npm:normalize',
+    'npm:tabzilla:img',
```
Comment (pdehaan) [inline gulpfile.js]: Nit: move tabzilla tasks closer together

File: src/scripts/main.js [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+  data.append('email', emailVal);
+
+  var xhr = new XMLHttpRequest();
+  xhr.open('POST', '/signup', true);
+  xhr.onload = function () {
+  // do something to response
+    console.log(this.responseText);
+  };
+  xhr.send(data);
+})
```
Comment (pdehaan) [inline src/scripts/main.js]: Missing semi

Comment (johngruen) [inline src/scripts/main.js]: Arg

File: src/scripts/main.js [hunk 8]
```diff
@@ -0,0 +1,14 @@
+document.querySelector('#signup').addEventListener('click', function(ev) {
+  ev.preventDefault();
+  var data = new FormData();
+  var emailVal = document.querySelector('#email').value;
+  data.append('email', emailVal);
+
+  var xhr = new XMLHttpRequest();
+  xhr.open('POST', '/signup', true);
+  xhr.onload = function () {
+  // do something to response
```
Comment (pdehaan) [inline src/scripts/main.js]: Nit: hawkward indenting (need to check if ESLint is linting this).

**Comment by johngruen** [comment]
adding `semi: [2, "always"]` to eslint

**Comment by johngruen** [comment]
@pdehaan i don't think eslint is checking comment indentation, but just look at the possibilities: http://eslint.org/docs/rules/

File: views/index.jade [hunk 9]
```diff
@@ -1,41 +1,29 @@
...
+    #splash-cta.wrapper
+      header
+        h1.splash-title= mainHeader
+        h2.splash-desc= subHeader
+      form(id="signupForm", role="form", class="splash-email-submit" novalidate="")
+        .success.email-submitted.hidden= mailResponses.success
+        .error.invalid-email.hidden= mailResponses.invalid
+        .error.server-error.hidden= mailResponses.serverError
+        input(type="email" id="email", name="email", placeholder="what's your email?" tabindex="1")
+        input(type="submit" id="signup" value="Sign up")
```
Comment (pdehaan) [inline views/index.jade]: Move placeholder text and button text into the data.json file?
Can be a future PR.

Comment (johngruen) [inline views/index.jade]: @pdehaan fuck it, going back in

**Comment by pdehaan** [comment]
Lookin great, @johngruen!
Feel free to merge away when ready.

**Comment by pdehaan** [comment]
Wait, sass errors in Travis. Sorry, trying to review on iPad

**Comment by johngruen** [comment]
@pdehaan scss-lint gripes about the following
```
_base.scss:13 [W] PropertySortOrder: Properties should be ordered background, outline
_layout.scss:29 [W] PlaceholderInExtend: Prefer using placeholder selectors (e.g. %some-placeholder) with @extend
_theme.scss:1 [W] IdSelector: Avoid using id selectors
modules/_footer.scss:1 [W] IdSelector: Avoid using id selectors
modules/_splash.scss:1 [W] IdSelector: Avoid using id selectors
modules/_splash.scss:26 [W] IdSelector: Avoid using id selectors
modules/_splash.scss:39 [W] IdSelector: Avoid using id selectors
modules/_splash.scss:44 [W] IdSelector: Avoid using id selectors
modules/_splash.scss:80 [W] IdSelector: Avoid using id selectors
```

none of which are terrible...might be that changing normalize and tabzilla to scss makes Travis angry?

**Comment by johngruen** [comment]
@pdehaan didn't notice that first scss-lint error with all the ones below it...that might have been causing the problem. Also, i think we can put a pin in this for the evening. Thanks for being the best!

**Comment by pdehaan** [comment]
I... I have no idea what's going on.
I cloned your branch and can run it locally 10 times and get 4 different results. Almost always fails. I'm starting to think that there are some deep timing issues with Gulp or our tasks, and it's trying to run the Sass `@import` statements before the files are copied into the `/src/vendor/` directory.

```sh
➜  idea-town git:(issue-25) ✗ gulp build
[22:22:51] Using gulpfile ~/dev/github/idea-town/gulpfile.js
[22:22:51] Starting 'build'...
[22:22:51] Starting 'clean'...
[22:22:51] Finished 'clean' after 2.94 ms
[22:22:51] Starting 'vendor'...
[22:22:51] Starting 'npm:tabzilla:img'...
[22:22:51] Finished 'vendor' after 5.01 ms
[22:22:51] Starting 'scripts'...
[22:22:51] Finished 'build' after 31 ms
[22:22:52] gulp-notify: [Gulp notification] Scripts task complete
[22:22:52] Finished 'scripts' after 186 ms
[22:22:52] Starting 'styles'...
[22:22:52] Finished 'npm:tabzilla:img' after 220 ms
[22:22:52] Starting 'npm:tabzilla:css'...
Error in plugin 'sass'
Message:
    src/styles/main.scss
  4:9  file to import not found or unreadable: ../vendor/normalize.css/normalize
Current dir:
[22:22:52] Finished 'styles' after 35 ms
[22:22:52] Starting 'images'...
[22:22:52] gulp-notify: [Gulp notification] Images task complete
[22:22:52] Finished 'npm:tabzilla:css' after 28 ms
[22:22:52] Starting 'npm:normalize'...
[22:22:52] gulp-notify: [Gulp notification] Images task complete
[22:22:52] Finished 'npm:normalize' after 22 ms
[22:22:52] gulp-notify: [Gulp notification] Images task complete
[22:22:52] gulp-notify: [Gulp notification] Images task complete
[22:22:52] Finished 'images' after 56 ms
[22:22:52] Starting 'prep-config'...
[22:22:52] Finished 'prep-config' after 18 ms
```

```sh
➜  idea-town git:(issue-25) ✗ gulp build
[23:33:04] Using gulpfile ~/dev/github/idea-town/gulpfile.js
[23:33:04] Starting 'build'...
[23:33:04] Starting 'clean'...
[23:33:04] Finished 'clean' after 3.17 ms
[23:33:04] Starting 'vendor'...
[23:33:04] Starting 'npm:tabzilla:img'...
[23:33:04] Finished 'vendor' after 4.98 ms
[23:33:04] Starting 'scripts'...
[23:33:04] Finished 'build' after 39 ms

stream.js:94
      throw er; // Unhandled stream error in pipe.
            ^
Error: ENOENT, stat '/Users/pdehaan/dev/github/idea-town/public/scripts/main.js'
```

```sh
➜  idea-town git:(issue-25) ✗ gulp build
[23:33:08] Using gulpfile ~/dev/github/idea-town/gulpfile.js
[23:33:08] Starting 'build'...
[23:33:08] Starting 'clean'...
[23:33:08] Finished 'clean' after 3.1 ms
[23:33:08] Starting 'vendor'...
[23:33:08] Starting 'npm:tabzilla:img'...
[23:33:08] Finished 'vendor' after 4.92 ms
[23:33:08] Starting 'scripts'...
[23:33:08] Finished 'build' after 26 ms
[23:33:09] gulp-notify: [Gulp notification] Scripts task complete
[23:33:09] Finished 'scripts' after 185 ms
[23:33:09] Starting 'styles'...
[23:33:09] 'npm:tabzilla:img' errored after 197 ms
[23:33:09] Error: ENOENT, stat '/Users/pdehaan/dev/github/idea-town/public/vendor/mozilla-tabzilla/media/img/tabzilla-static-high-res.png'
ENOENT, stat '/Users/pdehaan/dev/github/idea-town/public/vendor/mozilla-tabzilla/media/img/tabzilla-static-high-res.png'
ENOENT, stat '/Users/pdehaan/dev/github/idea-town/public/vendor/mozilla-tabzilla/media/img/tabzilla-static-high-res.png'
Error in plugin 'sass'
Message:
    src/styles/main.scss
  4:9  file to import not found or unreadable: ../vendor/normalize.css/normalize
Current dir:
[23:33:09] Finished 'styles' after 16 ms
```

**Comment by pdehaan** [comment]
Ah, OK, I think I got it... [`del()`](https://www.npmjs.com/package/del) and [`run-sequence()`](https://www.npmjs.com/package/run-sequence) both take callbacks, which let you signal when the task is "done":

### gulpfile.js snippet:
```
gulp.task('clean', function (done) {
  return del([
    'public',
    'src/vendor'
  ], done);
});

gulp.task('vendor', function (done) {
  return runSequence([
    'npm:tabzilla:img',
    'npm:tabzilla:css',
    'npm:normalize'
  ], done);
});

gulp.task('build', function (done) {
  return runSequence(
    'clean',
    'vendor',
    'scripts',
    'styles',
    'images',
    'prep-config',
    done
  );
});
```
(See full [gulpfile.js](https://gist.github.com/pdehaan/ce59f486077faec1e24f) in this gist.)

### Output:
```
➜  idea-town git:(issue-25) ✗ gulp build
[23:56:33] Using gulpfile ~/dev/github/idea-town/gulpfile.js
[23:56:33] Starting 'build'...
[23:56:33] Starting 'clean'...
[23:56:33] Finished 'clean' after 20 ms
[23:56:33] Starting 'vendor'...
[23:56:33] Starting 'npm:tabzilla:img'...
[23:56:33] Starting 'npm:tabzilla:css'...
[23:56:33] Starting 'npm:normalize'...
[23:56:33] Finished 'npm:normalize' after 13 ms
[23:56:33] Finished 'npm:tabzilla:css' after 15 ms
[23:56:33] Finished 'npm:tabzilla:img' after 21 ms
[23:56:33] Finished 'vendor' after 22 ms
[23:56:33] Starting 'scripts'...
[23:56:33] gulp-notify: [Gulp notification] Scripts task complete
[23:56:33] Finished 'scripts' after 166 ms
[23:56:33] Starting 'styles'...
[23:56:33] gulp-notify: [Gulp notification] Styles task complete
[23:56:33] Finished 'styles' after 206 ms
[23:56:33] Starting 'images'...
[23:56:33] gulp-notify: [Gulp notification] Images task complete
[23:56:33] gulp-notify: [Gulp notification] Images task complete
[23:56:33] gulp-notify: [Gulp notification] Images task complete
[23:56:33] gulp-notify: [Gulp notification] Images task complete
[23:56:33] Finished 'images' after 57 ms
[23:56:33] Starting 'prep-config'...
[23:56:33] Finished 'prep-config' after 19 ms
[23:56:33] Finished 'build' after 495 ms
```

Anyways, hopefully that unblocks you. If not, we can dig into it further in the morning.

**Comment by pdehaan** [comment]
@johngruen This looks awesome, great job!
Is this ready to merge, or are you still tweakin'?

**Comment by johngruen** [comment]
@pdehaan ready...want to do the honors?

**Comment by pdehaan** [comment]
:+1:

**Comment by pdehaan** [comment]
You guys are awesome!

**Pull request merged**